angular-template.js

```diff
@@ -7,5 +7,28 @@ module.exports = {
             Overrides to Angular template recommended rules:
             https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin-template/src/configs/recommended.json
         */
+
+        /*
+            Overrides to Angular rules outside of the recommended configuration:
+        */
+
+        '@angular-eslint/template/accessibility-alt-text': 'error',
+        '@angular-eslint/template/accessibility-elements-content': 'error',
+        '@angular-eslint/template/accessibility-label-for': 'error',
+        '@angular-eslint/template/accessibility-label-has-associated-control': 'error',
+        '@angular-eslint/template/accessibility-table-scope': 'error',
+        '@angular-eslint/template/accessibility-valid-aria': 'error',
+        '@angular-eslint/template/click-events-have-key-events': 'error',
+        '@angular-eslint/template/conditional-complexity': 'error',
+        '@angular-eslint/template/cyclomatic-complexity': 'error',
+        '@angular-eslint/template/i18n': 'error',
+        '@angular-eslint/template/mouse-events-have-key-events': 'error',
+        '@angular-eslint/template/no-any': 'error',
+        '@angular-eslint/template/no-autofocus': 'error',
+        '@angular-eslint/template/no-call-expression': 'error',
```

Comment:
This is over kill in our case. Yes, we need to consider performance and use memoization and pipes in many cases. However, there are also many other cases where a function call has no impact on performance—where a 1ms execution is imperceivable for users.

Reply:
Can you provide an example of a violation of this rule? The docs don't have any and I don't think I understand what they mean by "Disallows calling expressions in templates, except for output handlers". Aren't all function calls, property accesses, operators, etc known as "expressions"? Does this mean your template can only be dumb static code? Clearly I'm missing something.

Is any of this code illegal?
```html
<div *ngIf="step === 'amount' || (step === 'card' && transaction.action !== 'iframe-sent' && transaction.action !== 'html-provided' && transaction.action !== 'redirect-to-url')">
  Enter your card details
</div>
<div *ngIf="canShowCardStep">
  Enter your card details
</div>
```

Reply:
They're referring function calls, likely because these hide complexity and tend to be more complex. However, if you're like me you'd put your first NgIf into a function call, and it would still execute at <1ms. I keep track of the number of conditionals and try to make these reactive with setters, and I definitely draw the line at array iteration.

The second primary factor to CD performance is the number of components on the page. This is why we added support  to material side-navs for lazy-loading and full page viewing ([not full and animations broken](https://stackblitz.com/edit/angular-qaqyoc)). This is a trick I stole from mobile, like iOS, which allows you to build vertical navigation without routing, destroy invisible views, and drop your component count to nothing FOR FREE. This is why I hate side-nav implementations that use floats to try to be performant. It's not, and they're missing the point (bit of rambling—super interesting topic).

Reply:
@TrevorKarjanis this is ready to close